packages/xstate-wallet/src/messaging.ts

```diff
@@ -134,17 +134,22 @@ export class MessagingService implements MessagingServiceInterface {
     const {id: requestId} = request;
 
     switch (request.method) {
+      case 'EnableEthereum':
+        if (this.store.chain.ethereumIsEnabled) {
+          window.parent.postMessage(jrs.success(requestId, {success: true}), '*');
+        } else {
+          this.eventEmitter.emit('AppRequest', {type: 'ENABLE_ETHEREUM', requestId});
+          // TODO: @tom should it send {success: true} at this point?
```

Comment:
No - the workflow would need to do that, otherwise it would return success even if the user doesn't approve. If we decide to return the selected address, then we won't need to change the existing workflow.